src/twisted/test/test_ftp.py

```diff
@@ -36,8 +37,75 @@
 
 
 
+def _has_ipv6():
```

Comment:
`test_conch` and `test_udp` both already define functions with *exactly* this name, and almost exactly this same implementation. 3 total repetitions of this code is too many :).  Please refactor it into a common (private, for now) location, so we're not making 3 different useless sockets during the test suite.